Classes/Projection/ProjectionManager.php

```diff
@@ -182,11 +183,20 @@ public function catchUp(string $projectionIdentifier): int
      */
     private function getHighestAppliedSequenceNumber(string $fullProjectionIdentifier): int
     {
-        $cacheId = md5($fullProjectionIdentifier);
-        if (!$this->projectionCache->has($cacheId)) {
-            return 0;
+        if (!isset($this->projections[$fullProjectionIdentifier])) {
```

Comment:
Why not change the signature to expect an instance of `Projection` instead? In the calling side we have it anyways and `getProjection()` is pretty "expensive"